bin/loadmeta.jl

```diff
@@ -61,42 +61,17 @@ function load_versions(dir::String)
 end
 
-const JULIA_VERSIONS = Dict(
-    v"0.1.1"  => Version("bddb70374fdbb2256782398c13430fa4d8b55d0b"),
-    v"0.1.2"  => Version("1187040d027210cc466f0b6a6d54118fd692cf2d"),
-    v"0.2.0"  => Version("05c6461b55d9a66f05ead24926f5ee062b920d6b"),
-    v"0.2.1"  => Version("e44b5939057d87c1e854077108a1a6d66203f4fa"),
-    v"0.3.0"  => Version("768187890c2709bf2ff06818f40e1cdc79bd44b0"),
-    v"0.3.1"  => Version("c03f413bbdb46c00033f4eaad402995cfe3b7be5"),
-    v"0.3.2"  => Version("8227746b95146c2921f83d2ae5f37ecd146592d8"),
-    v"0.3.3"  => Version("b24213b893ace6ca601722f6cdaac15f2b034b7b"),
-    v"0.3.4"  => Version("33920266908e950936ef3f8503810a2d47333741"),
-    v"0.3.5"  => Version("a05f87b79ad62beb033817fdfdefa270c9557aaf"),
-    v"0.3.6"  => Version("0c24dca65c031820b91721139f0291068086955c"),
-    v"0.3.7"  => Version("cb9bcae93a32b42cec02585c387396ff11836aed"),
-    v"0.3.8"  => Version("79599ada444fcc63cb08aed64b4ff6a415bb4d29"),
-    v"0.3.9"  => Version("31efe690bea1fd8f4e44692e205fb72d34f50ad3"),
-    v"0.3.10" => Version("c8ceeefcc1dc25953a644622a895a3adcbc80dad"),
-    v"0.3.11" => Version("483dbf5279efe29f94cf35d42de0ab0a673fb34e"),
-    v"0.3.12" => Version("80aa77986ec84067f4696439f10fc77c93703bc7"),
-    v"0.4.0"  => Version("0ff703b40afddf9b705bd6a06d3a59cb4c089ea5"),
-    v"0.4.1"  => Version("cbe1bee3a8f27ba4f349556857bf4615ee5fa68f"),
-    v"0.4.2"  => Version("bb73f3489d837e3339fce2c1aab283d3b2e97a4c"),
-    v"0.4.3"  => Version("a2f713dea5ac6320d8dcf2835ac4a37ea751af05"),
-    v"0.4.4"  => Version("ae683afcc4cc824dcf68251569a6cf8760e4fb6f"),
-    v"0.4.5"  => Version("2ac304dfba75fad148d4070ef4f8a2e400c305bb"),
-    v"0.4.6"  => Version("2e358ce975029ec97aba5994c17d4a2169c3b085"),
-    v"0.4.7"  => Version("ae26b25d43317d7dd3ca05f60b70677aab9c0e08"),
-    v"0.5.0"  => Version("3c9d75391c72d7c32eea75ff187ce77b2d5effc8"),
-    v"0.5.1"  => Version("6445c82d0060dbe82b88436f0f4371a4ee64d918"),
-    v"0.5.2"  => Version("f4c6c9d4bbbd9587d84494e314f692c15ff1f9c0"),
-    v"0.6.0"  => Version("903644385b91ed8d95e5e3a5716c089dd1f1b08a"),
-    v"0.6.1"  => Version("0d7248e2ff65bd6886ba3f003bf5aeab929edab5"),
-    v"0.6.2"  => Version("d386e40c17d43b79fc89d3e579fc04547241787c"),
-    v"0.7.0"  => Version("a4cb80f3edcf8cea00bd9660e3b65f544f41462f"),
-    v"1.0.0"  => Version("5d4eaca0c9fa3d555c79dbacdccb9169fdf64b65"),
-    v"1.1.0"  => Version("6ef1d76c50a39c0ce68b4e42948a1499c1551415"), # dummy commit, not actual 1.1
-)
+const JULIA_VERSIONS = Dict{VersionNumber,Version}()
+for line in eachline(`git -C $juliadir ls-remote --tags origin`)
+    global JULIA_VERSIONS
+    # Lines are in the form 'COMMITSHA\trefs/tags/TAG'
+    sha, ref = split(line, '\t')
+    _, _, tag = split(ref, '/')
+    occursin(Base.VERSION_REGEX, tag) || continue
+    ver = VersionNumber(tag)
+    JULIA_VERSIONS[ver] = Version(sha)
+end
+JULIA_VERSIONS[v"1.1.0"] = Version("6ef1d76c50a39c0ce68b4e42948a1499c1551415") # dummy commit, not actual 1.1
```

Comment:
If you do this first then when the real version is released, the correct tree hash will replace it.